### ISSUE TITLE
iedit doesn't respect restricted range

### ISSUE THREAD

**Issue by darkfeline** [syl20bnr/spacemacs#6223]
#### Description :octocat:

iedit doesn't respect restricted range.

#### Reproduction guide :beetle:
- Start Emacs
- Create Python file with content:
~~~
def foo(spam):
    return spam
          
def bar(spam):
    return spam
~~~      
- Use `* r r e` on `spam` to restrict range to function and start iedit.
- Replace `spam` with, e.g., `eggs`

*Observed behaviour:* :eyes:
`spam` in both functions are changed.

*Expected behaviour:* :broken_heart:
Only `spam` in one function is changed.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.20
- Spacemacs branch: update-layer-doc (rev. 9f0a983)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(ivy
 (auto-completion :variables auto-completion-enable-sort-by-usage t)
 better-defaults gtags ibuffer spell-checking syntax-checking semantic typography
 (chrome :variables edit-server-new-frame nil)
 dash elfeed
 (shell :variables shell-default-shell 'shell shell-default-term-shell
        (mir/with-executable "/usr/local/bin/fish" "/usr/bin/fish" "/bin/bash" "/bin/sh")
        shell-default-height 50)
 version-control git c-c   common-lisp csv dockerfile go html javascript latex lua markdown python scheme shell-scripts sql yaml mir-magit-annex mir-pass mir-rc mir-rc-qualia mir-rc-mail)
```

**Comment by bmag** [comment]
I can confirm the bug. The function `ahs-to-iedit` in `spacemacs-editing-visual/packages.el` starts Iedit, but doesn't restrict Iedit according to AHS's range:
```elisp
(defun ahs-to-iedit ()
  (interactive)
  (cond
   ((and (not (eq dotspacemacs-editing-style 'emacs))
         (configuration-layer/package-usedp 'evil-iedit-state))
    (evil-iedit-state/iedit-mode))
   ((and (eq dotspacemacs-editing-style 'emacs)
         (configuration-layer/package-usedp 'iedit))
    (iedit-mode))
   (t (ahs-edit-mode t))))
```

AHS seems to have 3 ranges: whole buffer, function, and "display area". Currently `ahs-to-iedit` works well for the whole bufffer case. For the function case, `iedit-restrict-function` may be helpful, although it's not guaranteed that AHS and Iedit always agree on what is the current function (but it's probably good enough). I am not sure what "display area" means.

Fortunately, the beginning and end of the active range are easily obtained by `(ahs-current-plugin-prop 'start)` and `(ahs-current-plugin-prop 'end)`, and can be passed to `iedit-restrict-region`.
Patch on the way.

**Comment by darkfeline** [comment]
"display area" means what is displayed in the window currently.  There's a small gotcha that starting `ahs` decreases the window height due to the transient state popup, but otherwise works as advertised.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
